marlowe-runtime/web-server/Language/Marlowe/Runtime/Web/Server/DTO.hs

```diff
@@ -408,6 +426,73 @@ instance IsCardanoEra era => FromDTO (TxBody era) where
     where
       asType = AsTxBody $ cardanoEraToAsType $ cardanoEra @era
 
+instance HasDTO (Tx era) where
+  type DTO (Tx era) = Web.TextEnvelope
+
+instance IsCardanoEra era => ToDTO (Tx era) where
+  toDTO = toDTO . serialiseToTextEnvelope Nothing
+
+instance IsCardanoEra era => FromDTO (Tx era) where
+  fromDTO = hush . deserialiseFromTextEnvelope asType <=< fromDTO
+    where
+      asType = AsTx $ cardanoEraToAsType $ cardanoEra @era
+
+newtype ShelleyTxWitness era = ShelleyTxWitness (TxWitness (ShelleyLedgerEra era))
+
+instance HasTypeProxy era => HasTypeProxy (ShelleyTxWitness era) where
+    data AsType (ShelleyTxWitness era) = AsShelleyTxWitness (AsType era)
+    proxyToAsType _ = AsShelleyTxWitness (proxyToAsType (Proxy :: Proxy era))
+
+instance
+  ( HasTypeProxy era
+  , Typeable (ShelleyLedgerEra era)
+  , ValidateScript (ShelleyLedgerEra era)
+  , Ledger.Core.Script (ShelleyLedgerEra era) ~ Ledger.Alonzo.Scripts.Script (ShelleyLedgerEra era)
+  ) => SerialiseAsCBOR (ShelleyTxWitness era) where
+
+  serialiseToCBOR (ShelleyTxWitness wit) = Binary.serialize' wit
+
+  deserialiseFromCBOR _ bs = do
+    let
+      lbs = BSL.fromStrict bs
+
+      annotator :: forall s. Binary.Decoder s (Binary.Annotator (TxWitness (ShelleyLedgerEra era)))
+      annotator = Binary.fromCBOR
+
+    (w :: TxWitness (ShelleyLedgerEra era)) <- Binary.decodeAnnotator "Shelley Tx Witness" annotator lbs
+    pure $ ShelleyTxWitness w
+
+instance
+  ( IsShelleyBasedEra era
+  , ValidateScript (ShelleyLedgerEra era)
+  , Ledger.Core.Script (ShelleyLedgerEra era) ~ Ledger.Alonzo.Scripts.Script (ShelleyLedgerEra era)
+  ) => HasTextEnvelope (ShelleyTxWitness era) where
+  textEnvelopeType _ = do
+    "ShelleyTxWitness:" <> case shelleyBasedEra :: ShelleyBasedEra era of
+       ShelleyBasedEraAlonzo -> "Alonzo"
+       ShelleyBasedEraBabbage -> "Babbage"
```

Comment:
This will result in `"ShelleyTxWitness:Alonzo"` or `"ShelleyTxWitness:Babbage"`. It think the convention for parameterized types would be to replace the colon with a space (`ShelleyTxWitness Alonzo`).